packages/web3/src/signer/tx-builder.ts

```diff
@@ -113,16 +115,16 @@ export abstract class TransactionBuilder {
     return { ...response, groupIndex: response.fromGroup, gasPrice: fromApiNumber256(response.gasPrice) }
   }
 
-  async buildUnsignedTx(params: SignUnsignedTxParams): Promise<Omit<SignUnsignedTxResult, 'signature'>> {
-    const data = { unsignedTx: params.unsignedTx }
-    const decoded = await this.nodeProvider.transactions.postTransactionsDecodeUnsignedTx(data)
+  buildUnsignedTx(params: SignUnsignedTxParams): Omit<SignUnsignedTxResult, 'signature'> {
+    const decoded = unsignedTxCodec.decode(hexToBinUnsafe(params.unsignedTx))
+    const [fromGroup, toGroup] = groupIndexOfTransaction(decoded)
     return {
-      fromGroup: decoded.fromGroup,
-      toGroup: decoded.toGroup,
+      fromGroup: fromGroup,
+      toGroup: toGroup,
       unsignedTx: params.unsignedTx,
-      txId: decoded.unsignedTx.txId,
-      gasAmount: decoded.unsignedTx.gasAmount,
-      gasPrice: fromApiNumber256(decoded.unsignedTx.gasPrice)
+      txId: UnsignedTxCodec.txId(decoded),
```

Comment:
The `UnsignedTx.txId` will encode the tx. Maybe we can directly calculate the tx id using the raw tx bytes.

Reply:
Fixed, thanks :+1: